controllers/baremetalset_controller.go

```diff
@@ -535,27 +573,40 @@ func (r *BaremetalSetReconciler) deleteOwnerRefLabeledObjects(instance *ospdirec
 	return nil
 }
 
-// Get the IP from the management network for a particular BaremetalHost in the BaremetalSet
-func findNewMgmtIP(instance *ospdirectorv1beta1.BaremetalSet, bmhName string) (string, error) {
-	// First check if we already have an IP for this host
-	if bmhStatus, ok := instance.Status.BaremetalHosts[bmhName]; ok {
-		if bmhStatus.MgmtIP != "" {
-			return bmhStatus.MgmtIP, nil
-		}
+// Get the IP, gateway and netmask from the management network for a particular BaremetalHost in the BaremetalSet
+func findIPs(instance *ospdirectorv1beta1.BaremetalSet, bmhName string) (ip string, gateway string, netmask string, err error) {
+	_, network, err := net.ParseCIDR(instance.Spec.MgmtCIDR)
+
+	if err != nil {
+		return "", "", "", err
 	}
 
-	_, network, err := net.ParseCIDR(instance.Spec.MgmtCIDR)
+	// Gateway will just be first address in the network for now
+	gw, err := cidr.Host(network, 1)
```

Comment:
we may need to revisit this as I'm not sure Management networks would be set as a default gateway. If the network is just used for accessing machines they may not need a default GW/route on that network. For now as this is the only way to get to the machine we can use it though.